### PR TITLE
feat(user_roles): Add accept invitation API and `UserJWTAuth`

### PR DESCRIPTION
**Pull request by ThisIsMani**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

This PR will add a new API for accepting invitation and also a new JWT auth type which has only `user_id`.

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

To support the Accept Invite flow in the following flow.
```mermaid
flowchart TD
    A["Connect Account"] -- Email --> B{"User Exists"}
    B -- No --> C["Create new \n Org \n Merchant \n User \n User Role"]
    C --> D["Send Verify Email"]
    D --> E["Verify Email Token"]
    B -- Yes --> D
    E --> H{"Is there\n any perferred \nmerchant"}
    F -- ==0 --> I{{"Send list of \nmerchants which\n user has access to\n along with an\n intermediate token"}}
    F -- >1 --> G[Select the first role with active status]
    H -- Yes --> M["Send the token \nwith that merchant_id"]
    H -- No --> F{"How many \n active merchants\n does user have \naccess to"}
    I -- merchant_ids, \nintermediate_token --> O["Accept Invite"]
    subgraph Accept Invite
        O --> P["Change status to active"]
        P --> G
        G --> M
    end
    K["Sign In"] -- Email, Password --> H
```

## How did you test it?

```curl
curl --location 'http://localhost:8080/user/user/invite/accept' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer JWT' \
--data '{
    "merchant_ids": [
         "merchant_id1",
         "merchant_id2",
         "merchant_id3"
    ],
	"need_dashboard_entry_response": true
}'
```
If any of the `merchant_id` status is `invitation_sent`, then you will be getting the following response.
```
{
    "token": "JWT with merchant_id, user_id, user_role",
    "merchant_id": "merchant_id",
    "name": "user name",
    "email": "user email",
    "verification_days_left": null,
    "user_role": "user role"
}
```
If `need_dashboard_entry_response` is `false` or not sent, then the response will be 200 OK.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by racnan — COMMENTED** [review]
Lets give this API the possibility to not send the sign payload as response payload. Maybe query params?

File: crates/router/src/routes/app.rs [hunk 1]
```diff
@@ -921,6 +921,7 @@ impl User {
             .service(web::resource("/role/list").route(web::get().to(list_roles)))
             .service(web::resource("/role/{role_id}").route(web::get().to(get_role)))
             .service(web::resource("/user/invite").route(web::post().to(invite_user)))
+            .service(web::resource("/user/accept_invite").route(web::post().to(accept_invitation)))
```
Comment (racnan) [inline crates/router/src/routes/app.rs:924]: combine these

File: crates/router/src/services/authentication.rs [hunk 2]
```diff
@@ -276,6 +300,33 @@ pub async fn get_admin_api_key(
         .await
 }
 
+#[derive(Debug)]
+pub struct UserJWTAuth;
```
Comment (racnan) [inline crates/router/src/services/authentication.rs:304]: rename

**Review by apoorvdixit88 — APPROVED** [review]

File: crates/router/src/routes/app.rs [hunk 3]
```diff
@@ -922,6 +922,7 @@ impl User {
             .service(web::resource("/role").route(web::get().to(get_role_from_token)))
             .service(web::resource("/role/{role_id}").route(web::get().to(get_role)))
             .service(web::resource("/user/invite").route(web::post().to(invite_user)))
+            .service(web::resource("/user/invite/accept").route(web::post().to(accept_invitation)))
```
Comment (racnan) [inline crates/router/src/routes/app.rs:925]: Good to have: Scope `/user`

**Review by racnan — APPROVED** [review]
Check if `need_dashboard_entry_response` needs to be in query param.

**Review by harsh-sharma-juspay — APPROVED** [review]

**Pull request merged**